src/EditorFeatures/CSharpTest/Wrapping/ChainedExpressionWrappingTests.cs

```diff
@@ -0,0 +1,317 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.CSharp.Wrapping;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Wrapping
+{
+    public class ChainedExpressionWrappingTests : AbstractWrappingTests
+    {
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+            => new CSharpWrappingCodeRefactoringProvider();
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsWrapping)]
+        public async Task TestMissingWithSyntaxError()
+        {
+            await TestMissingAsync(
+@"class C {
+    void Bar() {
+        [||]the.quick().brown.fox(,);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsWrapping)]
+        public async Task TestMissingWithoutEnoughChunks()
+        {
+            await TestMissingAsync(
+@"class C {
+    void Bar() {
+        [||]the.quick();
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsWrapping)]
+        public async Task TestWithEnoughChunks()
+        {
+            await TestAllWrappingCasesAsync(
```

Comment:
❔ Should this also include the wrap-without-alignment case?

Reply:
yup.  i'm updating these as we speak :)